### PR TITLE
Fix table not reacting to type change

### PR DESCRIPTION
**Pull request by vitaliyworks**

### CODE REVIEW

File: packages/konsole-app/src/panels/RecordPanel.tsx [hunk 1]
```diff
@@ -54,6 +54,7 @@ export const RecordPanel = () => {
   const [records, setRecords] = useState<IRecord[]>([]);
   const [search, setSearch] = useState('');
   const [delayedSearch, setDelayedSearch] = useState(search);
+  const filteredRecords = records.filter(record => recordType === record.type || !recordType);
```
Comment (wachulski) [inline packages/konsole-app/src/panels/RecordPanel.tsx:57]: this shouldn't go here, but in the RegistryClient's `queryRecords`

**Review by wachulski — CHANGES_REQUESTED** [review]
Clever change, but those lines break encapsulation of records querying with filtering built-in. This piece should just consume what RegistryClient offers.

File: packages/konsole-app/src/panels/RecordPanel.tsx (hunk 1 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx:57]: But RegistryClient will make an unnecessary network call. I think it should only be responsible for low level querying

Comment (wachulski) [reply packages/konsole-app/src/panels/RecordPanel.tsx:57]: nope;
1) when you have thousands or millions (whatever) of records do you still want to fetch full size of dataset
2) in each client you'd need to replicate the logic of filtering, ordering, paging....

Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx]: OK, I see that if paging is supported on the server side, we want to do this differently.

File: packages/konsole-app/src/config.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import * as config from '@dxos/config';
 
 // Generated by webpack-version-file-plugin.
 // eslint-disable-next-line import/no-unresolved
-import buildInfo from '../version.json';
+import buildInfo from '../dist/version.json';
```
Comment (richburdon) [inline packages/konsole-app/src/config.ts:9]: I don't think you should depend on build artifacts.

File: packages/konsole-app/src/panels/RecordPanel.tsx [hunk 3]
```diff
@@ -49,20 +49,19 @@ const delay = 500;
...
   const [search, setSearch] = useState('');
   const [delayedSearch, setDelayedSearch] = useState(search);
+  const recordTypesWithAll = [{ type: 'all', label: 'all' }, ...recordTypes];
 
   function refreshData() {
-    setRecordTypes([]);
-    setRecords([]);
     registryClient.getRecordTypes().then(setRecordTypes);
-    registryClient.queryRecords().then(setRecords);
+    registryClient.queryRecords(recordType === 'all' ? undefined : { type: recordType }).then(setRecords);
```
Comment (richburdon) [inline packages/konsole-app/src/panels/RecordPanel.tsx:61]: 'all' should not be used as a constant anywhere -- this is a bug. In this case fix the component so that type is undefined.

Comment (richburdon) [reply packages/konsole-app/src/panels/RecordPanel.tsx:61]: Also please use await consistently not then.

File: packages/konsole-app/src/panels/RecordPanel.tsx [hunk 4]
```diff
@@ -49,20 +49,19 @@ const delay = 500;
 export const RecordPanel = () => {
   const classes = useStyles();
   const registryClient = useRegistryClient();
-  const [recordType, setRecordType] = useState<string>('');
+  const [recordType, setRecordType] = useState<string>('all');
   const [recordTypes, setRecordTypes] = useState<IRecordType[]>([]);
   const [records, setRecords] = useState<IRecord[]>([]);
   const [search, setSearch] = useState('');
   const [delayedSearch, setDelayedSearch] = useState(search);
+  const recordTypesWithAll = [{ type: 'all', label: 'all' }, ...recordTypes];
```
Comment (richburdon) [inline packages/konsole-app/src/panels/RecordPanel.tsx:57]: Why is this required?

**Review by richburdon — CHANGES_REQUESTED** [review]

File: packages/konsole-app/src/config.ts (hunk 2 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/config.ts:9]: But version.json is generated by webpack either way, this just changes the location and prevents version.json from being committed with each change.

PS: Sorry, this change should not be in this PR, I will rebase on #106

File: packages/konsole-app/src/panels/RecordPanel.tsx (hunk 3 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx:61]: Ok, I will try to make the component should accept undefined, and not use 'all' for that.
Will use await.

File: packages/konsole-app/src/panels/RecordPanel.tsx (hunk 4 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx:57]: 'all' is needed in case user wants to see all the records, not just of specific types. Without this part, there would be no way to do that.  But I will think how to remove this 'all' constant

File: packages/konsole-app/src/panels/RecordPanel.tsx (hunk 3 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx]: Done

File: packages/konsole-app/src/panels/RecordPanel.tsx (hunk 4 above)
Comment (vitaliyworks) [reply packages/konsole-app/src/panels/RecordPanel.tsx]: Done

**Review by wachulski — APPROVED** [review]
